### ISSUE TITLE
Custom around on store to Wrap all the pages

### ISSUE THREAD

**Issue by gustavopvasconcellos** [vtex-apps/store-discussion#264]
**What are you trying to accomplish? Please describe.**  
Hello I need to, make a better performance on pages, develop a custom around on app VTEX Store at `store` block without content to can be replaced by any block on using "parent" behavior.

Using this, we can change this block on our app to create a custom context. 

If necessary, I can do a PR to repo ;)

**What have you tried so far?**  
I think there isn't nothing better to do.

**Additional info**  
We tried on a app to use this behavior and the requests had not call again on changing pages.

![image](https://user-images.githubusercontent.com/49173685/78601448-2e10c800-782b-11ea-870a-c8ca6646dd85.png)

On each pages extended by `store.list` all the around components maintain their values, not calling again.

**Comment by gustavopvasconcellos** [comment]
@klzns Initially, we need this context to add a Provider that does a request and on individual blocks we use its Consumer. All of this on our app.

The evolution on vtex.store is just the addition of a custom-around.

**Comment by gustavopvasconcellos** [comment]
@klzns What do you think about it?

**Comment by klzns** [comment]
Are fetching using GraphQL? If so, the query should be cached and it will not fetch it again.

**Comment by gustavopvasconcellos** [comment]
@klzns Yeah, I'm using GraphQL. But even if the requests shows up on Network tab we can guarantee this?

If I not use a Context to this, on my case, I will do the same request a lot of times, it's because of this I will need a Context.

My case is create custom shelfs to do a request to Chaordic. This requests returns to me X shelfs to be placed on different parts of some pages (store.home, store.search, store.orderplaced, store.not-found). 

Basically, my plan is the Context do the request and on each interface block have a schema to configurate what shelf will be shown.

**Comment by gustavopvasconcellos** [comment]
@klzns Vou explicar em portugues pois acho que vai ser mais facil...

Praticamente é por causa da API de Vitrines de recommendações da Chaordic. https://docs.linximpulse.com/v0-api-vitrines/docs/get-pages-recommendations

Ela me traz um array de vitrines com varios produtos cada uma, minha ideia é ter esse contexto para poder chamar a Query somente uma vez, e disponibilizar para cada bloco das vitrines, na home e outras páginas, o mesmo resultado da Query

Entendo que se eu fizer a Query direto no componente, mesmo sendo a mesma Query e tendo cache, seria uma requisição a mais para cada vitrine que fosse ser exibida

Foi dai que tive a ideia desse contexto, mas para isso, preciso de um around para coloca-lo e não acho que o `challenge` seja o around apropriado para isso.

**Comment by gustavopvasconcellos** [comment]
@klzns Acho que encontrei uma otima solução.
